client/src/game/components/GameBoard/GameBoard.tsx

```diff
@@ -132,8 +139,70 @@ export class GameBoard extends React.Component<GameBoard.Props, GameBoard.State>
     }
   }
 
-  private optimisticallyCalculateNextBoard(moveIntent: MoveIntent): number[] {
-    return this.props.gameState.board // TODO: calculate desired GameState from moveIntent
+  private optimisticallyCalculateNextBoard(moveIntent: MoveIntent, source: ChessboardLib.Square, target: ChessboardLib.Square): number[] {
+    const chess = new Chess();
+    const moveCount: number = this.props.gameState.moveCount;
+    if(moveCount === 0){
+      chess.clear();
+    }
+    else{
+      chess.load(this.getChessboardPosition());
+    }
+    
+    var board: number[] = this.props.gameState.board;
+
+    if (chess.move({ from: source, to: target }) != null) {
```

Comment:
Now I'm actually thinking we might not want the validation client side, since it somewhat reduces the utility of the `MoveValidator` @bennettj12  wrote. I'm going to update this PR to not do the validation for now, but leave it as a separate PR after we discuss.

Reply:
I'm doing a 180 on my previous concern. Client-side validation is a nice feature, but `MoveValidator` is still the source of truth.